src/app/components/bibliography/bibliography.component.ts

```diff
@@ -0,0 +1,163 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { BibliographicCitation, BibliographyParserService } from '../../services/xml-parsers/bibliography-parser.service';
+
+@Component({
+  selector: 'evt-bibliography',
+  templateUrl: './bibliography.component.html',
+  styleUrls: ['./bibliography.component.scss'],
+})
+export class BibliographyComponent implements OnDestroy {
+
+  public biblCits: BibliographicCitation[] = [];
+  private subscription: Subscription;
+  public isComplete: boolean;
+  selectedAlphOrder: 'Ascendent';
+  selectedField: 'Author';
+  selectedStyle: 'Chicago';
+  features: string[] = ['Author', 'Title', 'Date'];
+  styles: string[] = ['Chicago', 'APA'];
+  alphOrders: string[] = ['Ascendent', 'Descendent'];
+
+  constructor(
+    public bps: BibliographyParserService,
+    public http: HttpClient,
+  ) {
+
+    this.subscription = this.bps.getBibliographicCitations().subscribe(response => {
+      this.biblCits = response.citations;
+      this.isComplete = response.areComplete;
+    });
+    this.swap();
+    this.isComplete
+      ? this.sortBy('Ascendent', this.selectedField)
+      : this.sortBy('Ascendent');
+  }
+
+  swap() {
+    const biblPars = document.getElementsByClassName('biblPar');
+
+    const authors = document.getElementsByClassName('authors');
+    const dates = document.getElementsByClassName('dates');
+    const titles = document.getElementsByClassName('titles');
+    const editors = document.getElementsByClassName('editors');
+    const pubPlaces = document.getElementsByClassName('pubPlaces');
+    const publishers = document.getElementsByClassName('publishers');
+    const biblScopes = document.getElementsByClassName('biblScopes');
+    const series = document.getElementsByClassName('series');
+
+    const analyticAuthors = document.getElementsByClassName('analytic_author');
+    const analyticTitles = document.getElementsByClassName('analytic_titles');
+    const monogrAuthors = document.getElementsByClassName('monogr_authors');
+    const monogrTitles = document.getElementsByClassName('monogr_titles');
+    const monogrEditors = document.getElementsByClassName('monogr_editors');
+    const monogrImprintPubPlaces = document.getElementsByClassName('monogr_imprint_pubPlaces');
+    const monogrImprintPublishers = document.getElementsByClassName('monogr_imprint_publishers');
+    const monogrImprintDates = document.getElementsByClassName('monogr_imprint_dates');
+    const monogrBiblScopes = document.getElementsByClassName('monogr_biblScopes');
+    const seriesTitles = document.getElementsByClassName('series_titles');
+    const seriesBiblScopes = document.getElementsByClassName('series_biblScopes');
+
+    const notes = document.getElementsByClassName('notes');
+
+    const ps = Array.from(biblPars);
+    let a: HTMLCollection[];
```

Comment:
Name a to something meaningful

Reply:
You should not retrieve data from XML source within the component! Move this function in a dedicated parser outside the BibliographyComponent.